Add io.github.halfmexican.Mingle

halfmexican opened the pull request:
# Mingle 

A simple GTK4 app to play with Google's Emoji Kitchen.


### Please confirm your submission meets all the criteria

<!-- Please replace each `[ ]` by `[X]` when the step is complete -->

- [X] Please describe your application briefly.
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/halfmexican/mingle**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

bbhtt commented on com.github.halfmexican.Mingle.json:
```diff
@@ -0,0 +1,59 @@
+{
+    "app-id" : "com.github.halfmexican.Mingle",
```
It should be io.github, the rules are right there in the PR checklist.

halfmexican replied:
I feel so dumb, sorry

halfmexican replied:
changed the entire App ID

bbhtt commented on com.github.halfmexican.Mingle.json:
```diff
@@ -0,0 +1,59 @@
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/halfmexican/mingle.git",
+                    "branch": "main"
```
Use a tag and a commit, branch should not be used.

bbhtt commented on com.github.halfmexican.Mingle.metainfo.xml.in:
This should be upstream

hfiguiere replied:
it is upstream. This one which is `.in` should just be removed.

bbhtt commented on com.github.halfmexican.Mingle.metainfo.xml.in:
```diff
@@ -0,0 +1,18 @@
+    <content_attribute id="drugs-alcohol">mild</content_attribute>
+    <content_attribute id="drugs-tobacco">mild</content_attribute>
+    <content_attribute id="sex-themes">mild</content_attribute>
+    <content_attribute id="language-humor">mild</content_attribute>
```
Missing screenshots, releases https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/

hfiguiere replied:
also missing developer info (`developer_name` or `devloper` tags) and `url`.

halfmexican replied:
I've removed this file from the pr but added all the requested info and ran `appstreamcli validate`

hfiguiere commented on com.github.halfmexican.Mingle.metainfo.xml.in:
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop">
+	<id>com.github.halfmexican.Mingle.desktop</id>
```
id use the reversed dns names NEVER ends in `.desktop`

hfiguiere replied:
this must be fixed upstream when you fix the app-id.

halfmexican replied:
fixed!

hfiguiere commented on com.github.halfmexican.Mingle.json:
```diff
@@ -0,0 +1,59 @@
+    "runtime-version" : "46",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.vala"
```
I don't think this is necessary. The GNOME sdk contains Vala.

halfmexican replied:
removed

halfmexican commented:
1. I have [changed the app-id](https://github.com/halfmexican/mingle/pull/14) everywhere were needed upstream and tested that the application still built. 
2. added [releases](https://github.com/halfmexican/mingle/commit/fdc45b0e977e10f110d65944b2dda3db60603930), [screenshots](https://github.com/halfmexican/mingle/commit/fdaacaecea3131725643680250006ed58d3d7c1f), developer, and branding colors to metainfo and removed the file from this pr
3. removed the vala extension and used a commit and tag instead of branch

hfiguiere commented:
bot, build io.github.halfmexican.Mingle

hfiguiere commented on io.github.halfmexican.Mingle.json:
```diff
@@ -0,0 +1,57 @@
+        "--device=dri",
+        "--socket=wayland"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/vala/bin",
+        "prepend-ld-library-path" : "/usr/lib/sdk/vala/lib"
+    },
```
```suggestion
```

since you removed the SDK.

hfiguiere commented on io.github.halfmexican.Mingle.json:
```diff
@@ -0,0 +1,57 @@
+                    "type" : "git",
+                    "url" : "https://github.com/halfmexican/mingle.git",
+                    "commit": "fdaacaecea3131725643680250006ed58d3d7c1f",
+                    "tag":  "v0.10"
```
this tag doesn't exist.

halfmexican replied:
I'm so sorry for my incompetence.
it should exist [now](https://github.com/halfmexican/mingle/releases/tag/v0.10)

bbhtt commented:
bot, build io.github.halfmexican.Mingle
